### ISSUE TITLE
Gather all component configurations

### ISSUE THREAD

**Issue by davidbrochart** [asphalt-framework/asphalt#71]
It would be nice to be able to gather the configuration of the whole application, from the root component and recursively to all sub-components. I'm especially interested in doing something like this at the command line:
```bash
asphalt run config.yaml --help-all
```
And that would print all possible parameters. Ideally, there would be an explanation for each parameter, but I guess that it would require each component to use Click somehow.

**Comment by agronholm** [comment]
An interesting idea, but would probably require substantial effort to implement properly.

**Comment by davidbrochart** [comment]
If you go with Pydantic, we could take advantage of the fields description to display help about configuration:
```py
from pydantic import BaseModel, Field

class Foo(BaseModel):
    a: int = Field(description="a field")

Foo.__fields__["a"].field_info.description
# shows 'a field'
```

**Comment by agronholm** [comment]
Yes, that was part of the plan. I wonder if it would be better to just abstract out the API so it could accommodate other validation frameworks too?

**Comment by davidbrochart** [comment]
Yes why not, it would be even better.
